docs/plans/2026-06-07-gated-release-pipeline.md

```
@@ -0,0 +1,425 @@
+```bash
+git add .github/workflows/deploy-prod.yml
+git commit -m "ci: deploy-prod becomes reusable deploy+smoke (no independent tag trigger)"
+```
+
+---
+
+## Task 3: Rewrite `release.yml` as the stage -> gate -> promote orchestrator
+
+Single entry on tag push. Jobs (with `needs:` ordering for fail-closed):
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)